### PR TITLE
statistics: make exponential backoff estimation more safe

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/23059 

Problem Summary:

In the mentioned issue, the exp backoff est meets the case that there are no matched stats. So the original code will panic since the default logic branch is for `length=4`.

### What is changed and how it works?

The default branch is `length==0` instead.

### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- `No release note`

### CODE REVIEW

File: statistics/testdata/integration_suite_out.json [hunk 1]
```diff
@@ -24,6 +24,10 @@
       [
         "IndexReader_6 1.36 root  index:IndexRangeScan_5",
         "└─IndexRangeScan_5 1.36 cop[tikv] table:exp_backoff, index:idx(a, b, c, d) range:[1 1 1 3,1 1 1 5], keep order:false"
+      ],
+      [
+        "IndexReader_6 0.00 root  index:IndexRangeScan_5",
```
Comment (winoros) [inline statistics/testdata/integration_suite_out.json:29]: It's actually zero data.

**Review by time-and-fate — APPROVED** [review]
/lgtm

**Pull request merged**